hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequest.java

```diff
@@ -468,10 +468,10 @@ protected OMResponse createErrorOMResponse(
   protected void addResponseToDoubleBuffer(long trxIndex,
       OMClientResponse omClientResponse,
       OzoneManagerDoubleBufferHelper omDoubleBufferHelper) {
-    if (omClientResponse != null) {
-      omClientResponse.setFlushFuture(
-          omDoubleBufferHelper.add(omClientResponse, trxIndex));
-    }
+    Preconditions.checkNotNull(omClientResponse,
+        "OMClientResponse cannot be null");
+    omClientResponse.setFlushFuture(
```

Comment:
This will solve error cases for which added to double buffer. But there are cases which is not added to double buffer itself for failure. eg: which returns failure directly...
org.apache.hadoop.ozone.om.request.volume.OMVolumeSetOwnerRequest#validateAndUpdateCache()
`    if (!setVolumePropertyRequest.hasOwnerName()) {
      omResponse.setStatus(OzoneManagerProtocolProtos.Status.INVALID_REQUEST)
          .setSuccess(false);
      return new OMVolumeSetOwnerResponse(omResponse.build());
    }`

Above case will resolve most but still cases are there not following the pattern.

Reply:
It need be fixed too.